### PR TITLE
Add a8ad

### PR DESCRIPTION
**Pull request by szk-fancs**
Add A8 ad network for amp-ad.
Corporation name: F@NCommunications,Inc

### CODE REVIEW

**Comment by szk-fancs** [comment]
Add A8 ad network for amp-ad.
Corporation name: F@NCommunications,Inc

**Comment by szk-fancs** [comment]
Ping @lannka, @zhouyx
Sorry for ping you. 
Please review PR.

File: 3p/integration.js [hunk 1]
```diff
@@ -644,7 +646,7 @@ export function validateAllowedEmbeddingOrigins(window, allowedHostnames) {
     hostname = parseUrl(getSourceUrl(window.document.referrer)).hostname;
   }
   for (let i = 0; i < allowedHostnames.length; i++) {
-    // Either the hostname is exactly as whitelisted…
+    // Either the hostname is exactly as whitelisted...
```
Comment (zhouyx) [inline 3p/integration.js]: revert change here.

File: ads/a8.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
```
Comment (zhouyx) [inline ads/a8.js]: 2017

File: ads/a8.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<!---
+Copyright 2016 The AMP HTML Authors. All Rights Reserved.
```
Comment (zhouyx) [inline ads/a8.md]: 2017

File: ads/a8.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+<amp-ad width="300" height="250"
+	type="a8"
+	data-aid="170223075073"
+	data-wid="001"
+	data-eno="01"
+	data-mid="s00000016751001031000"
+	data-mat="2TCGYR-17GNXU-3L92-64Z8X"
+	data-type="static"
+>
+</amp-ad>```
```
Comment (zhouyx) [inline ads/a8.md]: git won't recognize ``` in same line. new line instead

File: ads/a8.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+## Configuration
```
Comment (zhouyx) [inline ads/a8.md]: So `data-aid` is required, while other 5 params are optional. Let's document this here.

File: examples/ads.amp.html [hunk 6]
```diff
@@ -170,6 +171,18 @@
     </amp-ad>
   </div>
 
+  <h2 id="a8">A8</h2>
+<amp-ad width="300" height="250"
```
Comment (zhouyx) [inline examples/ads.amp.html]: indentation please

File: examples/ads.amp.html [hunk 7]
```diff
@@ -170,6 +171,18 @@
...
+	type="a8"
+	data-aid="170223075073"
+	data-wid="001"
+	data-eno="01"
+	data-mid="s00000016751001031000"
+	data-mat="2TCGYR-17GNXU-3L92-64Z8X"
+	data-type="static"
+>
```
Comment (zhouyx) [inline examples/ads.amp.html]: same line please

File: ads/a8.md (hunk 3 above)
Comment (zhouyx) [inline ads/a8.md]: 2017

File: ads/a8.js (hunk 2 above)
Comment (zhouyx) [inline ads/a8.js]: 2017

**Comment by szk-fancs** [comment]
@zhouyx Thank you for review! Oh...I'm so sorry. My code was not clean...
And I misunderstood it as a copyright notation of the project.
Because "year" is not usually updated.
I will fix it.

**Comment by szk-fancs** [comment]
@zhouyx 
Fixed it.
Please check it, whenever convenient for you!
 Thank you.

File: ads/a8.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+import {writeScript, validateData} from '../3p/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function a8(global, data) {
+  validateData(data, ['aid'], ['wid','eno','mid','mat','type']);
```
Comment (zhouyx) [inline ads/a8.js]: nit: space after `,`

File: ads/a8.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+	data-mid="s00000016751001031000"
+	data-mat="2TCGYR-17GNXU-3L92-64Z8X"
+	data-type="static">
+</amp-ad>
+```
+
+## Configuration
+
+Required:data-aid
+Optional:data-wid,data-eno,data-mid,data-mat,data-type
```
Comment (zhouyx) [inline ads/a8.md]: nit: Please view the doc on git. You will notice they are in the same line. Please fix

Comment (szk-fancs) [inline ads/a8.md]: How about writing like this.

Required parameter:

- data-aid

Optional parameters:

- data-wid
- data-eno
- data-mid
- data-mat
- data-type

Comment (zhouyx) [inline ads/a8.md]: looks great

**Comment by szk-fancs** [comment]
@zhouyx 

Thank you for your review! 
I fixed it.

It's my first PR to open source project.
But I realized that my consideration was not enough at all.

You are really good at me.
Thank you my first reviewer!

**Comment by zhouyx** [comment]
@szk-fancs Thanks for the Pull Request! You are very welcome! This PR looks great, Merged.

**Pull request merged**